validator/sawtooth_validator/networking/future.py

```diff
@@ -127,24 +127,26 @@ def get(self, correlation_id):
 
     def remove(self, correlation_id):
         try:
-            del self._futures[correlation_id]
+            return self._futures.pop(correlation_id)
```

Comment:
This looks like a separate change? Should this be in the next commit?

Reply:
Good catch - moved it to the later commit.